File: app/src/main/java/com/zaclimon/acetv/ui/settings/UserInfoPresenterImpl.java
package com.zaclimon.acetv.ui.settings;

/**
 * Concrete implementation of the {@link UserInfoPresenter} interface.
 *
 * @author zaclimon
 * Creation date: 04/07/17
 */

public class UserInfoPresenterImpl implements UserInfoPresenter {

    private UserInfoView mUserInfoView;

    /**
     * Default constructor
     * @param userInfoView the view which will receive the required callbacks.
     */
    public UserInfoPresenterImpl(UserInfoView userInfoView) {
        mUserInfoView = userInfoView;
    }

    @Override
    public void retrieveUserInfo() {
        AsyncRetrieveUserInfos asyncRetrieveUserInfos = new AsyncRetrieveUserInfos(mUserInfoView);
        asyncRetrieveUserInfos.execute();
    }

}

File: app/src/main/java/com/zaclimon/tsutaeru/ui/settings/userinfo/UserInfoPresenterImpl.kt
package com.zaclimon.tsutaeru.ui.settings.userinfo

/**
 * Concrete implementation of the [UserInfoPresenter] interface.
 *
 * @property userInfoView The view that will be used to display the user information
 *
 * @author zaclimon
 */
class UserInfoPresenterImpl(endpoint: String?, view: UserInfoView) : UserInfoPresenter {

    private val userEndpoint = endpoint
    private val userInfoView = view

    override fun retrieveUserInfo() {
        val asyncRetrieveUserInfo = AsyncRetrieveUserInfo(userEndpoint, userInfoView)
        asyncRetrieveUserInfo.execute()
    }

}